### PR TITLE
[SPARK-45938][INFRA] Add `utils` to the dependencies of the `core/unsafe/network_common` module in `module.py`

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr add `utils` to the dependencies of the `core/unsafe/network_common` module in `module.py` due to  `utils` is direct dependency of `core/unsafe/network_common`:

https://github.com/apache/spark/blob/7120e6b88f2327ffb71c4bca14b10b15aeb26c32/core/pom.xml#L102-L106

https://github.com/apache/spark/blob/ef93dc247844aca778deda06897a2759ad5eeea4/common/unsafe/pom.xml#L44-L48

https://github.com/apache/spark/blob/ef93dc247844aca778deda06897a2759ad5eeea4/common/network-common/pom.xml#L177-L181

### Why are the changes needed?
All UTs of `utils` module are still in the `core` module now. This pr is used to avoid the issue of not running unit tests in the `core` module when changing the `utils` module.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass GitHub Actions

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: dev/sparktestsupport/modules.py [hunk 1]
```diff
@@ -178,7 +178,7 @@ def __hash__(self):
 
 core = Module(
     name="core",
-    dependencies=[kvstore, network_common, network_shuffle, unsafe, launcher],
+    dependencies=[kvstore, network_common, network_shuffle, unsafe, launcher, utils],
```
Comment (LuciferYang) [inline dev/sparktestsupport/modules.py:181]: cc @zhengruifeng @HyukjinKwon ,  `utils` module is also a direct dependency of unsafe and network-common. Do we need to modify the dependencies of `unsafe` and `network-common`? Or is the current change sufficient?

Comment (zhengruifeng) [reply dev/sparktestsupport/modules.py:181]: > utils module is also a direct dependency of unsafe and network-common

let's also add this dependency

File: dev/sparktestsupport/modules.py [hunk 2]
```diff
@@ -113,6 +113,14 @@ def __hash__(self):
     ],
 )
 
+utils = Module(
```
Comment (LuciferYang) [inline dev/sparktestsupport/modules.py:116]: Moving it is because of 

image

Comment (zhengruifeng) [reply dev/sparktestsupport/modules.py:116]: yeah, this is python :)

**Review by zhengruifeng — APPROVED** [review]

File: dev/sparktestsupport/modules.py (hunk 1 above)
Comment (LuciferYang) [reply dev/sparktestsupport/modules.py:181]: done

**Comment by LuciferYang** [comment]
Thanks @zhengruifeng

**Pull request closed without merging**